lib/apiservers/engine/backends/container_proxy.go

```diff
@@ -422,12 +463,92 @@ func (c *ContainerProxy) StreamContainerLogs(name string, out io.Writer, started
 	return nil
 }
 
+// Stop will stop (shutdown) a VIC container.
+//
+// returns
+//	error
+func (c *ContainerProxy) Stop(vc *viccontainer.VicContainer, name string, seconds int, unbound bool) error {
+	defer trace.End(trace.Begin(vc.ContainerID))
+
+	if c.client == nil {
+		return InternalServerError("ContainerProxy.Stop failed to get a portlayer client")
+	}
+
+	//retrieve client to portlayer
+	handle, err := c.Handle(vc.ContainerID, name)
+	if err != nil {
+		return err
+	}
+
+	// we have a container on the PL side lets check the state before proceeding
+	// ignore the error  since others will be checking below..this is an attempt to short circuit the op
+	// TODO: can be replaced with simple cache check once power events are propagated to persona
+	running, err := c.IsRunning(vc)
+	if err != nil && IsNotFoundError(err) {
+		cache.ContainerCache().DeleteContainer(vc.ContainerID)
+		return err
+	}
+	if !running {
+		return nil
```

Comment:
We don't actually want to return that error.  We only wanted to check for two pre-conditions: 1) not found, 2) not running.  For any other case, we just drop down to the rest of the code.

Btw, this code, logically is the same as before.  I moved this function from container.go to container_proxy.go to segregate the swagger code for stop out of container.go.